File: src/main/java/com/timurisachenko/websocketchat/pubsub/RedisSubscriber.java
package com.timurisachenko.websocketchat.pubsub;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.timurisachenko.websocketchat.model.ChatMessage;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.data.redis.connection.Message;
import org.springframework.data.redis.connection.MessageListener;
import org.springframework.data.redis.core.RedisTemplate;
import org.springframework.data.redis.listener.ChannelTopic;
import org.springframework.data.redis.listener.RedisMessageListenerContainer;
import org.springframework.messaging.simp.SimpMessageSendingOperations;
import org.springframework.messaging.simp.SimpMessagingTemplate;
import org.springframework.stereotype.Service;

@Slf4j
@RequiredArgsConstructor
@Service
public class RedisSubscriber {
    private final ObjectMapper objectMapper = new ObjectMapper();
    private final SimpMessageSendingOperations messagingTemplate;

    public void sendMessage(String publishMessage) {
        try {
            ChatMessage chatMessage = objectMapper.readValue(publishMessage, ChatMessage.class);
            messagingTemplate.convertAndSend("/topic/chat/room/" + chatMessage.getRoomId(), chatMessage);
        } catch (Exception e) {
            log.error(e.getMessage());
        }
    }
}

File: src/main/kotlin/com/itime/sernachat/service/RedisSubscriber.kt
package com.itime.sernachat.service

import com.fasterxml.jackson.databind.ObjectMapper
import com.itime.sernachat.domain.CommonMessage
import org.slf4j.Logger
import org.springframework.messaging.simp.SimpMessageSendingOperations
import org.springframework.stereotype.Service


@Service
class RedisSubscriber(
    val logger: Logger,
    val objectMapper: ObjectMapper,
    val messagingTemplate: SimpMessageSendingOperations
) {
    fun sendMessage(publishMessage: String?) {
        try {
            val message: CommonMessage = objectMapper.readValue(publishMessage, CommonMessage::class.java)
            messagingTemplate.convertAndSend("/topic/chat/room/" + message.chatId, message)
        } catch (e: Exception) {
            logger.error(e.message)
        }
    }
}
